controllers/chat/getSignedChannelList.js

```diff
@@ -1,10 +1,18 @@
+const {StreamChat} = require('stream-chat');
 const {responseSuccess} = require('../../utils/Responses');
-const GetstreamSingleton = require('../../vendor/getstream/singleton');
+const {CHANNEL_TYPE} = require('../../helpers/constants');
 
 const MAX_ITERATIONS = 10;
 
-const __queryChannelBuilder = async (userId, limit, offset) => {
-  const client = GetstreamSingleton.getChatInstance();
+/**
+ *
+ * @param {StreamChat} client
+ * @param {string} userId
+ * @param {number} limit
+ * @param {number} offset
+ * @returns
+ */
+const __queryChannelBuilder = async (client, userId, limit, offset) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The change from using a singleton instance of `StreamChat` to creating a new instance every time the function is called could potentially lead to performance issues. If this function is called frequently, it might be more efficient to use a singleton pattern to ensure that only one instance of `StreamChat` is created and reused across multiple requests.

```diff
- const client = new StreamChat(process.env.API_KEY, process.env.API_SECRET);
+ const client = GetstreamSingleton.getChatInstance();
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
no, we need different user connection because of connectUser and disconnectUser.
This what generates error 400 sometimes